### PR TITLE
[IMP] mail: display guest in channel member list

### PR DESCRIPTION
**Pull request by seb-odoo**
Part of task-2664860

### CODE REVIEW

File: addons/mail/static/src/models/channel_member_view.js [hunk 1]
```diff
@@ -15,17 +15,17 @@ registerModel({
...
-            this.channelMember.partner.openChat();
+            this.channelMember.persona.partner.openChat();
         },
         /**
          * @private
          * @returns {Boolean}
          */
         _computeHasOpenChat() {
-            return this.channelMember.partner ? true : false;
+            return this.channelMember.persona.partner ? true : false;
```
Comment (oomsveta) [inline addons/mail/static/src/models/channel_member_view.js:28]: Note that MessageView has also a compute of this kind (`MessageView/hasAuthorOpenChat`). Both seem to be a bit redundant; perhaps the two fields could be merged in the future?

**Review by oomsveta — COMMENTED** [review]
Looks fine to me overall; I'm looking forward to seeing the whole code unified with Persona 🙂

File: addons/mail/static/src/models/channel_member.js [hunk 2]
```diff
@@ -10,30 +10,48 @@ registerModel({
...
          */
-        _computeName() {
-            if (!this.partner) {
-                return;
+        _computeChannelAsOnlineMember() {
+            if (this.persona.partner) {
+                return this.persona.partner.isOnline ? replace(this.channel) : clear();
+            }
+            if (this.persona.guest) {
+                return clear();
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/mail/static/src/models/channel_member.js:46]: The goal is to introduce `im_status` for guest shortly after, until then offline is better than online.

File: addons/mail/static/src/models/channel_member.js [hunk 3]
```diff
@@ -10,30 +10,48 @@ registerModel({
...
         },
         /**
          * @private
          * @returns {FieldCommand}
          */
-        _computeChannelAsOnlineMember() {
-            return this.partner && this.partner.isOnline ? replace(this.channel) : clear();
+        _computeChannelAsOfflineMember() {
+            if (this.persona.partner) {
+                return !this.persona.partner.isOnline ? replace(this.channel) : clear();
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/mail/static/src/models/channel_member.js:30]: Well offline is "not isOnline" so I feel the condition is in the the right format according to the field name. I feel "replace" is better always before "clear" in this kind of pattern.

**Review by oomsveta — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**